packages/core/src/indexing-store/utils/format.ts

```diff
@@ -1,43 +1,182 @@
-import { encodeAsText } from "@/utils/encoding.js";
-
+import type {
```

Comment:
Maybe rename this file to `encoding.ts`? Could also move it up a level and get rid of the `utils` folder.